File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.py
special = ['c=', 'c-', 'dz=', 'd-', 'lj', 'nj', 's=', 'z=']
s = input()
count = 0
i = 0

while i < len(s):
    matched = False
    for sp in special:
        if s[i:i+len(sp)] == sp:
            count += 1
            i += len(sp)
            matched = True
            break
    if not matched:
        count += 1
        i += 1

print(count)

File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.js
const special = ['c=', 'c-', 'dz=', 'd-', 'lj', 'nj', 's=', 'z='];
const s = require('fs').readFileSync(0).toString().trim();
let count = 0;
let i = 0;

while (i < s.length) {
    let matched = false;
    for (const sp of special) {
        if (s.slice(i, i+sp.length) === sp) {
            count++;
            i += sp.length;
            matched = true;
            break;
        }
    }
    if (!matched) {
        count++;
        i++;
    }
}

console.log(count);